module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,52 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed_bitmap;
+    for (size_t i = 0; i < height; ++i) {
+        uint8_t sequence_begin = 0;
+        for (size_t j = 0; j < width; ++j) {
+            if (bitmap[i][j] != bitmap[i][j + 1] || j == width - 1) {
```

Comment:
That is brilliant :) I will implement it in my code, however I'd rather try to use iterators :)